analyses/cell-type-wilms-tumor-06/notebook/04_annotation_Across_Samples_exploration.Rmd

```diff
@@ -0,0 +1,310 @@
+
+
+```{r path_to_query}
+sample_metadata_file <- file.path(repository_base, "data", "current", params$scpca_project_id, "single_cell_metadata.tsv")
+
+metadata <- read.table(sample_metadata_file, sep = "\t", header = TRUE)
+
+# Create a data frames of all annotations
+cell_type_df <- metadata$scpca_sample_id |>
+  purrr::map(
+    # For each sample_id, do the following:
+    \(sample_id) {
+
+      # Read in the Seurat object
+      srat <- readRDS(
+        file.path(result_dir, sample_id, paste0("02b-fetal_kidney_label-transfer_", sample_id, ".Rds"))
+      )
+
+      # Create and return a data frame from the Seurat object with relevant annotations
+      # this data frame will have four columns: barcode, sample_id, compartment, organ
+      data.frame(
+        compartment = srat$fetal_kidney_predicted.compartment,
+        organ = srat$fetal_full_predicted.organ, 
+        PTPRC = FetchData(object = srat, vars = "ENSG00000081237", layer = "counts"),
+        VWF = FetchData(object = srat, vars = "ENSG00000110799", layer = "counts")
+        ) |>
+        tibble::rownames_to_column("barcode") |>
+        dplyr::mutate(sample_id = sample_id)
+    }
+  ) |>
+  # now combine all dataframes to make one big one
+  dplyr::bind_rows()
+
+```
+
+### Output file
+
+The report will be saved in the `notebook` directory. 
+
+
+## Functions
+
+
+#### do_Table_Heatmap
+
+`do_Table_Heatmap` shows heatmap of counts of cells for combinations of two metadata variables
+
+- `first_group` is the name of the first metadata to group the cells
+- `last_group` is the name of the second metadata to group the cells
+
+```{r fig.width=10, fig.height=4, out.width='100%'}
+do_Table_Heatmap <- function(first_group, last_group ){
+  
+  df <- cell_type_df %>%
+  group_by( !!sym(first_group), !!sym(last_group))%>% 
+  summarise(Nb = n())  %>%
+  mutate(C = sum(Nb)) %>%
+  mutate(percent = Nb/C*100)
+
+  
+p1 <- ggplot(df, aes(x= !!sym(first_group), y = !!sym(last_group), fill = Nb)) +
+   geom_tile() +
+   scale_fill_viridis_c()  +
+   theme_bw() +
+   theme(text = element_text(size = 20)) +
+   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
+   guides(fill=guide_legend(title=" Number of \n cells"))
+
+p2 <- ggplot(df, aes(x= !!sym(first_group), y = !!sym(last_group), fill = percent)) +
+   geom_tile() +
+   scale_fill_viridis_c()  +
+   theme_bw() +
+   theme(text = element_text(size = 20)) +
+   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
+   guides(fill=guide_legend(title=" Percentage of \n cells"))
+
+p <- p1 + p2 + plot_layout(ncol = 2)
+return(p)
+}
+```
+
+#### do_Feature_mean
+
+`do_Feature_mean` shows heatmap of mean expression of a feature grouped by a metadata. 
+
+- `group.by` is the name of the metadata to group the cells
+- `feature` is the name of the gene to average the expression
+
+```{r fig.width=10, fig.height=4, out.width='100%'}
+do_Feature_mean <- function(group.by, feature ){
```

Comment:
If we end up keeping this function, it should also definitely take the data frame (`cell_type_df`) as a formal argument and not just assume it's defined.